File: pcaixa/src/Moeda.php
<?php

namespace Caixa;

class Moeda {
    private float $valor;
    private string $codigoMoeda;
    private array $cotacoes = [
        'BRL' => [
            'USD' => 0.2,
            'PYG' => 5
        ],
        'USD' => [
            'BRL' => 5,
            'PYG' => 25  
        ],
        'PYG' => [
            'USD' => 1/25, // (1/5)/5,
            'BRL' => 0.2
        ]
    ];
    private array $simbolos = [
        'BRL' => 'R$',
        'USD' => 'US$',
        'PYG' => '₲'
    ]; 
    
    public function __construct(float $valor, string $codigoMoeda)
    {
        $this->valor = $valor;
        $this->codigoMoeda = $codigoMoeda;
    }
    
    public function getCodigoMoeda(): string
    {
        return $this->codigoMoeda;
    }
    
    public function getValor(): string
    {
        return $this->valor;
    }

	public function converterPara(string $codigoMoeda){
	    if ($codigoMoeda == $this->codigoMoeda){
	        return $this->valor;
	    }
        $valorConvertido = 
        ($this->valor * $this->cotacoes[$this->codigoMoeda][$codigoMoeda]);  	          	    
	    return $valorConvertido;
	}
	
	public function exibir()
	{
	    $saida = $this->simbolos[$this->codigoMoeda] . ' ' . str_replace('.',',',number_format($this->valor,2));
	    return $saida;
	}
	
	public function somar(Moeda $moeda)
	{
	    $soma = $this->valor;
	    $soma += ($moeda->getValor() *  $this->cotacoes[$moeda->getCodigoMoeda()][$this->codigoMoeda]);
	    return $soma;
	}
}

File: jcaixa/src/jcaixa/Moeda.java
package jcaixa;
import java.util.HashMap;

class Moeda {
    private float valor;
    private String codigoMoeda;
    private HashMap<String,HashMap<String,Float>> cotacoes = new HashMap<>();
    private HashMap<String,String> simbolos = new HashMap<>();
    
    public Moeda(float valor, String codigoMoeda)
    {
    	HashMap<String,Float> cotacao;    	
        cotacao = new HashMap<>();
        cotacao.put("USD", (float) 0.2);
        cotacao.put("PYG", (float) 5);
    	this.cotacoes.put("BRL", cotacao);
    	cotacao = new HashMap<>();
        cotacao.put("BRL", (float) 5);
        cotacao.put("PYG", (float) 25);
    	this.cotacoes.put("USD", cotacao);
    	cotacao = new HashMap<>();
        cotacao.put("USD", (float) 1/25);
        cotacao.put("BRL", (float) 0.2);
    	this.cotacoes.put("PYG", cotacao); 
    	
    	simbolos.put("BRL","R$");
    	simbolos.put("USD","US$");
    	simbolos.put("PYG","₲");
        
        this.valor = valor;
        this.codigoMoeda = codigoMoeda;
                
    }
    
    public String getCodigoMoeda()
    {
        return this.codigoMoeda;
    }
    
    public float getValor()
    {
        return this.valor;
    }

	public float converterPara(String codigoMoeda){
        float valorConvertido = 
        (this.valor * (float) this.cotacoes.get(this.codigoMoeda).get(codigoMoeda));  	          	    
	    return valorConvertido;
	}
	
	public String exibir()
	{
	    String saida = this.simbolos.get(this.codigoMoeda) + " " + String.format("%.2f",this.valor);
	    return saida;
	}
	
	public float somar(Moeda moeda)
	{
	    float soma = this.valor;
	    
	    soma += (moeda.getValor() *  this.cotacoes.get(moeda.getCodigoMoeda()).get(this.codigoMoeda));
	    return soma;
	}
	
}
